modules/swagger-codegen/src/main/resources/typescript-angular2/variables.mustache

```diff
@@ -1,9 +1,9 @@
 import { {{{injectionToken}}} } from '@angular/core';
 
-export const BASE_PATH = new {{{injectionToken}}}('basePath');
+export const BASE_PATH = new {{{injectionToken}}}<String>('basePath');
```

Comment:
That's weird. Can you post a minimum working example which doesn't compile?
`string` is the TypeScript string type and String is the default JavaScript string type, which is why I am wondering if the underlying issue is something else.
See e.g. [here](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)